pkg/bff/gds.go

```diff
@@ -32,3 +41,95 @@ func ConnectGDS(conf config.DirectoryConfig) (_ gds.TRISADirectoryClient, err er
 	}
 	return gds.NewTRISADirectoryClient(cc), nil
 }
+
+// Lookup makes a request on behalf of the user to both the TestNet and MainNet GDS
+// servers, returning 1..2 results (e.g. either or both GDS responses). If no results
+// are returned, Lookup returns a 404 not found error. If one of the GDS requests fails,
+// the error is logged, but the valid response is returned. If both GDS requests fail,
+// a 500 error is returned. This endpoint passes through the response from GDS as JSON,
+// the result should contain a registered_directory field that identifies which network
+// the record is associated with.
+func (s *Server) Lookup(c *gin.Context) {
+	// Bind the parameters associated with the lookup
+	params := &api.LookupParams{}
+	if err := c.ShouldBindQuery(&params); err != nil {
+		log.Warn().Err(err).Msg("could not bind request with query params")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		return
+	}
+
+	// Ensure that we have either ID or CommonName
+	if params.ID == "" && params.CommonName == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("must provide either uuid or common_name in query params"))
+		return
+	}
+
+	// Create the LookupRequest, omit registered directory as it is assumed we're
+	// looking up the value for the directory we're making the request to.
+	req := &gds.LookupRequest{Id: params.ID, CommonName: params.CommonName}
+	errs := make([]error, 2)
+	results := make([]map[string]interface{}, 2)
+
+	// Execute the request in parallel to both the testnet and the mainnet
+	ctx, cancel := context.WithTimeout(c.Request.Context(), 25*time.Second)
+	var wg sync.WaitGroup
+	defer cancel()
+	wg.Add(2)
+
+	lookup := func(client gds.TRISADirectoryClient, idx int, name string) {
+		defer wg.Done()
+		var (
+			err error
+			rep *gds.LookupReply
+		)
+
+		if rep, err = client.Lookup(ctx, req); err != nil {
+			serr, _ := status.FromError(err)
+			if serr.Code() != codes.NotFound {
+				log.Error().Err(err).Str("network", name).Msg("GDS lookup unsuccessful")
+				errs[idx] = err
+			}
+			return
+		}
```

Comment:
I think that your suggestions are already implemented - my question relates to a data structure in the protocol buffers that are not currently being used (or implemented by GDS). 

To clarify:

For error case 1 ("bad value"): there are two cases that result in this:

1. InvalidArgument: the user specifies something that is not a UUID in the ID field or not a domain in the CommonName field -- currently the backend does no validation of this, it just does a lookup (the front-end does have validation of this though). 
2. NotFound: the users specifies an id or common name that is not in the directory service. 

In the case of error 1.2 (bad value because it is not in either the testnet or the mainnet database) - the BFF will return a 404 not found error, which the UI can display (the current behavior with just one GDS). That code happens later on. So I think achieves the bubble error up semantics that you're suggesting.

In error case 2 - the BFF is handling as follows - if only 1 of the GDS is down, then it returns a response and logs the error; if both are down it logs both errors and returns a 500 error to the user -- the idea being that the front end can say something like "something went very wrong, please contact support".